harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/application/ui/mvi/ViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package com.harmony.kotlin.application.ui.mvi
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Base class for all of our ViewModels to inherit from
+ */
+abstract class ViewModel<out VS : ViewState, in A : Action> : LifecycleAwareViewModel(), CoroutineScope {
+
+  /**
+   * Observable state of the view
+   */
+  abstract val viewState: StateFlow<VS>
+
+  /**
+   * Notify the ViewModel of an Action that occurred on the View
+   */
+  abstract fun onAction(action: A)
+
+  override val coroutineContext: CoroutineContext
+    get() = job + Dispatchers.Main
```

Comment:
It could be better to have the `Dispatcher` in the constructor with a default value as `Dispatcher.main`, in case we need to change it for testing purposes. 

@franmontiel what do you think?

Reply:
I agree